### PR TITLE
proxy: expose proxy configuration

### PR DESCRIPTION
**Pull request by xiang90**
This is a readonly feature for now. We will see the requirement from community to decide to make this also writable.

### CODE REVIEW

File: proxy/proxy.go [hunk 1]
```diff
@@ -40,10 +42,16 @@ type GetProxyURLs func() []string
...
 func NewHandler(t *http.Transport, urlsFunc GetProxyURLs, failureWait time.Duration, refreshInterval time.Duration) http.Handler {
-	return &reverseProxy{
+	p := &reverseProxy{
 		director:  newDirector(urlsFunc, failureWait, refreshInterval),
 		transport: t,
 	}
+
+	mux := http.NewServeMux()
+	mux.Handle("/", p)
+	mux.HandleFunc("/v2/config/local/proxy", p.configHandler)
```
Comment (yichengq) [inline proxy/proxy.go]: do we need "/v2/" prefix?

File: proxy/proxy.go [hunk 2]
```diff
@@ -62,3 +70,37 @@ func readonlyHandlerFunc(next http.Handler) func(http.ResponseWriter, *http.Requ
 		next.ServeHTTP(w, req)
 	}
 }
+
+func (p *reverseProxy) configHandler(w http.ResponseWriter, r *http.Request) {
```
Comment (yichengq) [inline proxy/proxy.go]: a simple test on this would be great

File: proxy/proxy.go (hunk 1 above)
Comment (xiang90) [inline proxy/proxy.go]: yes. we need.

**Comment by xiang90** [comment]
@yichengq test added.

**Comment by yichengq** [comment]
lgtm after travis passes

**Pull request merged**